image_builder/image_builder/git.py

```diff
@@ -1,41 +1,40 @@
 from pathlib import Path
+import shutil
 from typing import Optional
 from urllib.parse import urlsplit, quote
 
-from git import Repo
-
 from racetrack_commons.dir import project_root
 from racetrack_client.log.context_error import wrap_context, ContextError
 from racetrack_client.client_config.client_config import Credentials
 from racetrack_client.manifest import Manifest
 from racetrack_client.utils.shell import shell, CommandError, shell_output
 
 
-def fetch_repository(workspace: Path, manifest: Manifest, git_credentials: Optional[Credentials]) -> Path:
+def fetch_repository(workspace: Path, manifest: Manifest, git_credentials: Optional[Credentials]) -> tuple[Path, str]:
     """Clone git repository to local workspace directory and return fetched project path"""
     remote_url = git_remote_with_credentials(manifest.git.remote, git_credentials)
 
     with wrap_context('cloning git repo'):
         try:
-            shell(f'GIT_TERMINAL_PROMPT=0 git clone {remote_url} {workspace.resolve()}',
-                  workdir=project_root(), print_log=False)
+            if manifest.git.branch:
+                assert '"' not in manifest.git.branch, 'illegal characters in a branch name'
```

Comment:
I intuitively think it might be better to use `git check-ref-format --branch`. Check `man git check-ref-format`.

Reply:
Good point. However, I've just checked that 
```
git check-ref-format --branch "\"&&sl&&echo"
```
doesn't raise any alarm, but I don't want it to be injected in the next command in place of `--branch "{manifest.git.branch}"`

So I'll use both: your command and my manual validation against quotes in a branch name.

Reply:
Or actually the main reason is that I need to validate it against command injection, so I need to check it before running any command, including `git check-ref-format`